linker/Linker.Steps/MarkStep.cs

```diff
@@ -273,6 +282,127 @@ void MarkCustomAttributes (ICustomAttributeProvider provider)
 			}
 		}
 
+		void MarkUserDependency (MethodReference context, CustomAttribute ca)
+		{
+			if (ca.ConstructorArguments.Count == 1 && ca.ConstructorArguments[0].Value is string s) {
+				string member = null;
+				string type = null;
+				string[] signature = null;
+				TypeDefinition td = null;
+
+				var sign = s.IndexOf ('(');
```

Comment:
Move the string parsing into a helper method?

Reply:
would need to lift a lot of local variables as this is the helper method ;-)